File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/products/dto/ProductAndAuctionResponse.java
package com.portfolio.auctionmarket.domain.products.dto;

import com.portfolio.auctionmarket.domain.auctions.dto.AuctionResponse;
import com.portfolio.auctionmarket.domain.auctions.entity.Auction;
import com.portfolio.auctionmarket.domain.products.entity.Product;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductAndAuctionResponse {
    private ProductResponse productResponse;

    private AuctionResponse auctionResponse;

    public static ProductAndAuctionResponse from(Product entity) {
        return ProductAndAuctionResponse.builder()
                .productResponse(ProductResponse.from(entity))
                .auctionResponse(AuctionResponse.from(entity.getAuction()))
                .build();
    }
}

File: AuctionMarket/auction-domain/src/main/kotlin/org/example/auctioncommon/domain/product/dto/ProductAndAuctionResponse.kt
package org.example.auctioncommon.domain.product.dto

import org.example.auctioncommon.domain.auction.dto.AuctionResponse
import org.example.auctioncommon.domain.auction.dto.toAuctionResponse
import org.example.auctioncommon.domain.product.entity.Product


data class ProductAndAuctionResponse(
    private val productResponse: ProductResponse? = null,
    private val auctionResponse: AuctionResponse? = null
)

fun Product.toDto(): ProductAndAuctionResponse {
    return ProductAndAuctionResponse(
        productResponse = this.toProductResponse(),
        auctionResponse = this.auction.toAuctionResponse()
    )
}